_posts/2023-06-26-angular-standalone-components.md

```diff
@@ -0,0 +1,181 @@
+---
+title: "Angular standalone components"
+excerpt: "A simplified way to build Angular applications"
+tags: Web Angular standalone
+authors:
+- Ankit Pant
+header:
+  teaser: /assets/images/post/angular-standalone-components.png
+  teaser_alt: Angular standalone components
+category: Frontend
+---
+
+![](/assets/images/post/angular-standalone-components.png)
+
+# Introduction
+
+Until now, the developers had to add or update NgModules specifying which components, directives, pipes, etc are available to be used in templates but after the release of the standalone feature, you don't need to do it anymore in the NgModules. Angular [standalone components](https://angular.io/guide/standalone-components "https://angular.io/guide/standalone-components") are independent and self-contained building blocks in an Angular application which can be used to independently import the dependencies required for its functionality.  
+Standalone components are not mandatory and there are no strict rules for using them, however, their use is recommended by Angular architects especially for newly created components.
+
+To designate a component, directive, or a pipe as a standalone component, you have to mark it as `standalone: true`. The component does not have to be declared or imported in `NgModule`, and you can import all the dependencies directly in the component itself, thus eliminating the use of `NgModule` entirely.
+
+ ```typescript
+ @Component({
+   selector: '...',
+   templateUrl: '...',
+   styleUrls: ['...'],
+   standalone: true,
+   imports: [
+     NgIf,
+     NgFor,
+     AsyncPipe,
+     ...
+   ],
+ })
+ export class FooBarComponent {
+```
+
+
+# Generate a standalone component via the Angular CLI
+
+ ```
+ ng generate component --standalone account-statements
+```
+
+A component generated with `--standalone` flag is not added to `NgModule` and contains default imports such as `CommonModule` to get started with the component right away.
+
+# Adopt standalone components incrementally
+
+Standalone components can be seamlessly combined with existing modules. They are easy to implement, and can be adopted incrementally into an app. A `standalone` component, pipe or directive can also be imported into an existing `NgModule` configuration, applied incrementally for complex apps, so that you can ensure that there are no breaking changes.
+
+ ```typescript
+ @NgModule({
+   declarations: [...],
+   exports: [...]
+   imports: [
+     AccountDetailsComponent, // This is a standalone component
+     CommonModule,
+     RouterModule.forChild(routes),
+     .....
+   ],
+ })
+```
+
+# Migrate a component to a Standalone
+
+You can convert a component in one of the following ways:
+
+* Using the [schematics](https://angular.io/guide/standalone-migration#migrate-an-existing-angular-project-to-standalone "https://angular.io/guide/standalone-migration#migrate-an-existing-angular-project-to-standalone")
+
+* Manually
+
+## Migrate to Standalone via schematics
```

Comment:
I don't know where should I execute those commands? In the root of the project or in the folder of the module I want to migrate? As a suggestion, I think we should recommend having the code covered with unit tests and e2e tests before the migration, if it's possible.

Reply:
Yes, this should be executed in the root of an application and of course, the app developers are responsible for making sure that the application is already tested and works before the migration.
@tcorral Do you want me to add this information as well?

Reply:
Updated the document with your suggestion with the Prerequisites and a note for running the schematics command at the root of the project.